Originated historically at the earliest in
https://github.com/termux/termux-packages/pull/23474#issuecomment-2683389623,
but is character for character convergently identical to the independently written
upstream commit,
https://github.com/stefanor/pypy/commit/7509998bdec6c5dd39cfdf069d06b46b157f47df

--- a/rpython/translator/c/src/precommondefs.h
+++ b/rpython/translator/c/src/precommondefs.h
@@ -11,7 +11,7 @@
 /* Define on Darwin to activate all library features */
 #define _DARWIN_C_SOURCE 1
 /* These must be set to 64 to enable large file support on 32-bit systems. */
-#if defined(i386) || defined(__i386__) || defined(__i386) || defined(_M_IX86)
+#if defined(i386) || defined(__i386__) || defined(__i386) || defined(_M_IX86) || defined(__arm__)
   #define _FILE_OFFSET_BITS 64
   #define _LARGEFILE_SOURCE 1
 #endif
